network/server_discovery.go

```diff
@@ -25,7 +25,8 @@ func (s *DefaultServer) GetRandomBootnode() *peer.AddrInfo {
 	nonConnectedNodes := make([]*peer.AddrInfo, 0)
 
 	for _, v := range s.bootnodes.getBootnodes() {
-		if !s.HasPeer(v.ID) {
+		// don't use HasPeer()
+		if !s.IsConnected(v.ID) && v.ID != s.host.ID() {
```

Comment:
Any more comment for this modification?
I don't see any difference between `s.IsConnected` and `s.HasPeer`.

Reply:
`IsConnected` check libp2p connection, `HasPeer` check `PeerConnInfo` is exist, `PeerConnInfo` exists but the `connection` does not necessarily exist (this is bug,I need more time to fix it.)